core/impl/src/main/java/pt/webdetails/cdf/dd/model/inst/writer/cdfrunjs/components/amd/CdfRunJsDataSourceComponentWriter.java

```diff
@@ -165,8 +167,10 @@ private void renderBuiltinCdaDatasource( StringBuilder out, CdfRunJsDashboardWri
     String dsName = dataSourceComp.getName();
     addFirstJsProperty( out, PropertyName.DATA_ACCESS_ID, buildJsStringValue( dsName ), Writer.INDENT2 );
 
-    String refreshPeriod = dataSourceComp.tryGetPropertyBindingByName( PropertyValue.COMPONENT_REFRESH_RATE ) != null ? dataSourceComp.tryGetPropertyBindingByName( PropertyValue.COMPONENT_REFRESH_RATE ).getValue() : "";
-    addJsProperty( out, PropertyName.DATA_ACCESS_STREAM_REFRESH_PERIOD, buildJsStringValue( refreshPeriod ), Writer.INDENT2 );
+    String dataSourceType = dataSourceComp.tryGetAttributeValue( DATA_ACCESS_TYPE, null );
```

Comment:
I think that similar code to this should be being added to:
1. amd: https://github.com/webdetails/cde/blob/master/core/impl/src/main/java/pt/webdetails/cdf/dd/model/inst/writer/cdfrunjs/properties/CdfRunJsDataSourcePropertyBindingWriter.java#L172
2. legacy: https://github.com/webdetails/cde/blob/master/core/impl/src/main/java/pt/webdetails/cdf/dd/model/inst/writer/cdfrunjs/properties/CdfRunJsCdaDataSourcePropertyBindingWriter.java#L23

In that way, CDF components would not need to be calling `query.setPushEnabled(true);` — the data source metadata would already be available, side-by-side with all of the other data source properties, and so the Query class could read it directly from the received query definition...
This is how all other data source properties are handled.

Reply:
Makes sense... will review the code to make that change... and add some tests along the way.
👍

Reply:
@dcleao I've analyzed the code that writes the property bindings and it seems it is only used when there is a PropertyBinding to render, so it will not always render, and more specifically it won't render when just a streaming datasource is used along with a simple componente like a table.

Reply:
Justo to be more clear (to whoever happens to read this), the previous finding was incorrect... what is happening is that the amd dashboards are not using those binding writer classes.